pyspecdata/lmfitdata.py

```diff
@@ -266,8 +241,12 @@ def eval(self, taxis=None, set_what=None, set_to=None):
             newdata.set_error(self.fit_axis,
                     self.get_error(self.fit_axis))
         # }}}
-        newdata
-        newdata.data[:] = self.fitfunc(p, taxis).flatten()
+        param_dict = {}
+        for j in range(len(list(p))):
+            case = {list(self.parameter_names)[j]:list(p)[j]}
+            param_dict.update(case)
+        self.set_guess(param_dict)
```

Comment:
I don't understand this -- why are you setting the guess here?  Couldn't this just be eliminated?

Reply:
sometimes, when we `eval` we do this after manually setting the parameters to the guess, but sometimes, we do this after fitting.  I'm suggesting that only the line added after this comment should be needed.

Reply:
(remember that we have a separate `settoguess` function for exactly this purpose)

Reply:
so, to be clear, I'm suggesting removing this for loop and the subsequent `set_guess`

Reply:
https://jmfrancklab.slack.com/archives/CV553TPRV/p1682684466462009?thread_ts=1682650761.090889&cid=CV553TPRV